test/installer/test_apt_installer.py

```diff
@@ -22,13 +20,79 @@ def test_should_load_not_on_ubuntu(self, get_version):
 
     @patch('colcon_bundle.installer.apt.get_ubuntu_distribution_version')
     def test_should_load_on_ubuntu(self, _):
-        sys.modules['apt'] = mock.MagicMock()
-        apt = AptBundleInstallerExtension()
-        assert apt.should_load() == True
+        with patch.dict("sys.modules", apt=mock.MagicMock()):
+            apt = AptBundleInstallerExtension()
+            assert apt.should_load() == True
 
     @patch('colcon_bundle.installer.apt.get_ubuntu_distribution_version')
     def test_should_load_on_ubuntu_no_python3_apt(self, _):
-        sys.modules['apt'] = None
-        apt = AptBundleInstallerExtension()
-        with self.assertRaises(RuntimeError): apt.should_load()
+        with patch.dict("sys.modules", apt=None):
+            apt = AptBundleInstallerExtension()
+            with self.assertRaises(RuntimeError): apt.should_load()
+
+    def test_apt_add_to_install_list_splits_version_specifier(self):
```

Comment:
This is now covered because we are using the same path for both. If the version does not exist the `package.versions.get(version, default)` will return the default. I have verified this by manually testing against the python-apt library in Ubuntu.